### PR TITLE
Add bowl view mode in stitching

### PR DESCRIPTION
**Pull request by wujunkai166**

### CODE REVIEW

**Comment by wujunkai166** [comment]
@windyuan

**Comment by wujunkai166** [comment]
@windyuan , please review the new patch. Thanks.

**Comment by windyuan** [comment]
@wujunkai166  check CI build errors.

	04:20:43,281 INFO  - libtool: link: g++ -fPIC -DSTDC99 -W -Wall -D_REENTRANT -Wformat -Wformat-security -fstack-protector -g -DDEBUG -std=c++0x -I/usr/include/libdrm -I../ext/atomisp -I../xcore -I../modules -I/usr/include/opencv -g3 -o .libs/test-soft-image test_soft_image-test-soft-image.o -L/opt/media/install/lib ../modules/soft/.libs/libxcam_soft.so ../modules/ocl/.libs/libxcam_ocl.so /opt/media/build/libxcam/modules/interface/.libs/libxcam_interface.so -lOpenCL -lopencv_shape -lopencv_stitching -lopencv_superres -lopencv_videostab -lopencv_aruco -lopencv_bgsegm -lopencv_bioinspired -lopencv_ccalib -lopencv_cvv -lopencv_dnn -lopencv_dpm -lopencv_fuzzy -lopencv_hdf -lopencv_line_descriptor -lopencv_optflow -lopencv_plot -lopencv_reg -lopencv_saliency -lopencv_stereo -lopencv_structured_light -lopencv_rgbd -lopencv_surface_matching -lopencv_tracking -lopencv_datasets -lopencv_text -lopencv_face -lopencv_video -lopencv_ximgproc -lopencv_calib3d -lopencv_features2d -lopencv_flann -lopencv_xobjdetect -lopencv_objdetect -lopencv...
	04:20:43,284 INFO  - mv -f .deps/test_device_manager-test-device-manager.Tpo .deps/test_device_manager-test-device-manager.Po
	04:20:43,287 INFO  - /bin/sh ../libtool --tag=CXX --mode=link g++ -fPIC -DSTDC99 -W -Wall -D_REENTRANT -Wformat -Wformat-security -fstack-protector -g -DDEBUG -std=c++0x -I/usr/include/libdrm -ldrm -I../ext/atomisp -I../xcore -I../modules -I/usr/include/opencv -g3 -L/opt/media/install/lib -o test-device-manager test_device_manager-test-device-manager.o ../modules/ocl/libxcam_ocl.la -lopencv_shape -lopencv_stitching -lopencv_superres -lopencv_videostab -lopencv_aruco -lopencv_bgsegm -lopencv_bioinspired -lopencv_ccalib -lopencv_cvv -lopencv_dnn -lopencv_dpm -lopencv_fuzzy -lopencv_hdf -lopencv_line_descriptor -lopencv_optflow -lopencv_plot -lopencv_reg -lopencv_saliency -lopencv_stereo -lopencv_structured_light -lopencv_rgbd -lopencv_surface_matching -lopencv_tracking -lopencv_datasets -lopencv_text -lopencv_face -lopencv_video -lopencv_ximgproc -lopencv_calib3d -lopencv_features2d -lopencv_flann -lopencv_xobjdetect -lopencv_objdetect -lopencv_ml -lopencv_xphoto -lopencv_highgui -lopencv_videoio -lopencv_imgcodecs -lopenc...
	04:20:43,339 INFO  - test-image-stitching.cpp: In function 'int main(int, char**)':
	04:20:43,339 INFO  - test-image-stitching.cpp:328:78: error: cannot convert 'XCam::StitchResMode' to 'XCam::SurroundMode' for argument '6' to 'XCam::SmartPtr<XCam::CLImageHandler> XCam::create_image_360_stitch(const XCam::SmartPtr<XCam::CLContext>&, bool, XCam::CLBlenderScaleMode, bool, bool, XCam::SurroundMode, XCam::StitchResMode, int, bool)'
	04:20:43,339 INFO  - enable_fisheye_map, enable_lsc, res_mode, fisheye_num, all_in_one).dynamic_cast_ptr<CLImage360Stitch> ();
	04:20:43,339 INFO  - ^
	04:20:43,339 INFO  - test-image-stitching.cpp:328:113: error: expected primary-expression before '>' token
	04:20:43,339 INFO  - enable_fisheye_map, enable_lsc, res_mode, fisheye_num, all_in_one).dynamic_cast_ptr<CLImage360Stitch> ();
	04:20:43,339 INFO  - ^
	04:20:43,339 INFO  - test-image-stitching.cpp:328:116: error: expected primary-expression before ')' token
	04:20:43,339 INFO  - enable_fisheye_map, enable_lsc, res_mode, fisheye_num, all_in_one).dynamic_cast_ptr<CLImage360Stitch> ();
	04:20:43,339 INFO  - ^

**Comment by wujunkai166** [comment]
Solved. Forgot to update changes in tests. @windyuan

File: modules/ocl/cl_fisheye_handler.cpp [hunk 1]
```diff
@@ -374,44 +375,72 @@ CLFisheyeHandler::generate_fisheye_table (
...
-    XCAM_FAIL_RETURN (
-        ERROR, table_kernel->set_arguments (args, work_size) == XCAM_RETURN_NO_ERROR,
-        XCAM_RETURN_ERROR_CL, "kernel_fisheye_table set arguments failed");
-
-    XCAM_FAIL_RETURN (
-        ERROR, table_kernel->execute (table_kernel, true) == XCAM_RETURN_NO_ERROR,
-        XCAM_RETURN_ERROR_CL, "[%s] execute kernel_fisheye_table failed", get_name ());
-
+    if(_surround_mode == BowlView) {
+        BowlDataConfig bowl_data_config = get_bowl_config();
```
Comment (windyuan) [inline modules/ocl/cl_fisheye_handler.cpp]: too many lines there.
it's better change to 

    if (bowl_view)
           fill_bowl_view_table ();
    else
            fill_sphere_view_table ();

**Pull request merged**